src/element/resizeElements.ts

```diff
@@ -0,0 +1,405 @@
+            });
+          } else {
+            mutateElement(element, {
+              height,
+              ...adjustXYWithRotation("s", element, 0, deltaY, angle),
+            });
+          }
+          break;
+        }
+        case "e": {
+          const width = element.width + deltaX;
+
+          if (isLinearElement(element)) {
+            if (element.points.length > 2 && width <= 0) {
+              // Someday we should implement logic to flip the shape.
+              // But for now, just stop.
+              break;
+            }
+            mutateElement(element, {
+              width,
+              ...adjustXYWithRotation("e", element, deltaX, 0, angle),
+              points: rescalePoints(0, width, element.points),
+            });
+          } else {
+            mutateElement(element, {
+              width,
+              ...adjustXYWithRotation("e", element, deltaX, 0, angle),
+            });
+          }
+          break;
+        }
+        case "rotation": {
+          const [x1, y1, x2, y2] = getElementAbsoluteCoords(element);
+          const cx = (x1 + x2) / 2;
+          const cy = (y1 + y2) / 2;
+          let angle = (5 * Math.PI) / 2 + Math.atan2(y - cy, x - cx);
+          if (event.shiftKey) {
+            angle += SHIFT_LOCKING_ANGLE / 2;
+            angle -= angle % SHIFT_LOCKING_ANGLE;
+          }
+          if (angle >= 2 * Math.PI) {
+            angle -= 2 * Math.PI;
+          }
+          mutateElement(element, { angle });
+          break;
+        }
+      }
+
+      if (resizeHandle) {
+        resizeHandle = normalizeResizeHandle(element, resizeHandle);
+      }
+      normalizeDimensions(element);
+
+      document.documentElement.style.cursor = getCursorForResizingElement({
+        element,
+        resizeHandle,
+      });
+      mutateElement(el, {
+        x: element.x,
+        y: element.y,
+      });
+
+      return true;
+    } else if (selectedElements.length > 1) {
+      // EXPERIMENT: se only
+      if (resizeHandle === "se") {
+        const [x1, y1, x2, y2] = getCommonBounds(selectedElements);
+        const handleOffset = 4 / appState.zoom; // XXX import constant
+        const dashedLinePadding = 4 / appState.zoom; // XXX import constant
+        const minSize = handleOffset * 4;
+        const minScale = Math.max(minSize / (x2 - x1), minSize / (y2 - y1));
+        const scale = Math.max(
+          (x - handleOffset - dashedLinePadding - x1) / (x2 - x1),
+          (y - handleOffset - dashedLinePadding - y1) / (y2 - y1),
+        );
+        if (scale > minScale) {
+          selectedElements.forEach((element) => {
+            const width = element.width * scale;
+            const height = element.height * scale;
+            const x = element.x + (element.x - x1) * (scale - 1);
+            const y = element.y + (element.y - y1) * (scale - 1);
+            mutateElement(element, { width, height, x, y });
+          });
+        }
+        return true;
```

Comment:
@enzoferey This is what I'm doing now.

Reply:
Thanks for pointing. It will indeed collide with my work, but you separated well the case of single and multi element resizing, so no problem on my side 👍